Source: Fedora
Upstream: Fixed in rev172

diff --git a/libs/lensfun/CMakeLists.txt b/libs/lensfun/CMakeLists.txt
index ef3dcf0..253a08c 100644
--- a/libs/lensfun/CMakeLists.txt
+++ b/libs/lensfun/CMakeLists.txt
@@ -26,3 +26,8 @@ INSTALL(TARGETS lensfun
         RUNTIME DESTINATION ${BINDIR}
         LIBRARY DESTINATION ${LIBDIR}
         ARCHIVE DESTINATION ${LIBDIR})
+
+# pkgconfig support
+CONFIGURE_FILE(${CMAKE_CURRENT_SOURCE_DIR}/lensfun.pc.cmake ${CMAKE_CURRENT_BINARY_DIR}/lensfun.pc @ONLY)
+INSTALL( FILES ${CMAKE_CURRENT_BINARY_DIR}/lensfun.pc DESTINATION ${LIBDIR}/pkgconfig )
+
diff --git a/libs/lensfun/lensfun.pc.cmake b/libs/lensfun/lensfun.pc.cmake
new file mode 100644
index 0000000..6d14b12
--- /dev/null
+++ b/libs/lensfun/lensfun.pc.cmake
@@ -0,0 +1,13 @@
+prefix=@CMAKE_INSTALL_PREFIX@
+bindir=@CMAKE_INSTALL_PREFIX@/@BINDIR@
+libdir=@CMAKE_INSTALL_PREFIX@/@LIBDIR@
+includedir=@CMAKE_INSTALL_PREFIX@/@INCLUDEDIR@
+datadir=@CMAKE_INSTALL_PREFIX@/@DATADIR@
+docdir=@CMAKE_INSTALL_PREFIX@/@DOCDIR@
+
+Name: lensfun
+Description: A photographic lens database and access library
+Version: @VERSION_MAJOR@.@VERSION_MINOR@.@VERSION_MICRO@.@VERSION_BUGFIX@
+Requires.private: glib-2.0
+Libs: -L${libdir} -llensfun
+Cflags: -I${includedir} -I${includedir}/lensfun
