Source: Gentoo
Upstream: No
Reason: Remove ricer flags
Index: audacity-src-1.3.7/lib-src/sbsms/configure.in
===================================================================
--- audacity-src-1.3.7.orig/lib-src/sbsms/configure.in
+++ audacity-src-1.3.7/lib-src/sbsms/configure.in
@@ -123,14 +123,6 @@ else
 fi
 AM_CONDITIONAL(ENABLE_PORTAUDIO, test x$enable_portaudio = xyes)
 
-CXXFLAGS=`echo $CXXFLAGS | sed -e "s/-g//" -e "s/-O2//"`
-
-if test x$enable_debug = xyes; then
-   CXXFLAGS="$CXXFLAGS -g -O0"
-else
-   CXXFLAGS="$CXXFLAGS -O3 -fomit-frame-pointer"
-fi
-
 if test x$enable_static = xyes; then
    CXXFLAGS="$CXXFLAGS"
 else
Index: audacity-src-1.3.7/lib-src/sbsms/configure
===================================================================
--- audacity-src-1.3.7.orig/lib-src/sbsms/configure
+++ audacity-src-1.3.7/lib-src/sbsms/configure
@@ -15359,14 +15359,6 @@ else
 fi
 
 
-CXXFLAGS=`echo $CXXFLAGS | sed -e "s/-g//" -e "s/-O2//"`
-
-if test x$enable_debug = xyes; then
-   CXXFLAGS="$CXXFLAGS -g -O0"
-else
-   CXXFLAGS="$CXXFLAGS -O3 -fomit-frame-pointer"
-fi
-
 if test x$enable_static = xyes; then
    CXXFLAGS="$CXXFLAGS"
 else
