Source: Gentoo
Upstream: No
Reason: Missing header

diff -Naur raul-0.5.1-orig/src/SMFWriter.cpp raul-0.5.1/src/SMFWriter.cpp
--- raul-0.5.1-orig/src/SMFWriter.cpp	2008-09-09 11:39:22.000000000 -0600
+++ raul-0.5.1/src/SMFWriter.cpp	2009-07-30 00:20:55.590573854 -0600
@@ -19,6 +19,7 @@
 #include <cassert>
 #include <cstring>
 #include <iostream>
+#include <stdio.h>
 #include <glibmm/miscutils.h>
 #include <raul/SMFWriter.hpp>
 
diff -Naur raul-0.5.1-orig/tests/midi_ringbuffer_test.cpp raul-0.5.1/tests/midi_ringbuffer_test.cpp
--- raul-0.5.1-orig/tests/midi_ringbuffer_test.cpp	2008-09-09 11:39:22.000000000 -0600
+++ raul-0.5.1/tests/midi_ringbuffer_test.cpp	2009-07-30 00:22:49.542541333 -0600
@@ -2,6 +2,7 @@
 #include "raul/EventRingBuffer.hpp"
 #include <iostream>
 #include <cstring>
+#include <cstdio>
 #include "raul/midi_names.h"
 
 using namespace std;
