From 17a33964c32035f6b9986e13726ae3582a126244 Mon Sep 17 00:00:00 2001
From: Götz Waschk <goetz.waschk@gmail.com>
Date: Sun, 10 Feb 2013 18:54:17 +0000
Subject: Fix mixed declarations and code in WavPack support

---
diff --git a/src/wavpack_tag.c b/src/wavpack_tag.c
index 1437996..f5765d0 100644
--- a/src/wavpack_tag.c
+++ b/src/wavpack_tag.c
@@ -82,15 +82,15 @@ gboolean Wavpack_Tag_Write_File (FILE *file_in, gchar *filename_in, vcedit_state
  */
 gboolean Wavpack_Tag_Read_File_Tag (gchar *filename, File_Tag *FileTag)
 {
-    if (!filename || !FileTag)
-        return FALSE;
-
     WavpackContext *wpc;
     gchar *field, *field2;
     guint length;
 
     int open_flags = OPEN_TAGS;
 
+    if (!filename || !FileTag)
+        return FALSE;
+
     wpc = WavpackOpenFileInput(filename, NULL, open_flags, 0);
 
     if ( wpc == NULL ) {
@@ -271,9 +271,6 @@ gboolean Wavpack_Tag_Read_File_Tag (gchar *filename, File_Tag *FileTag)
 
 gboolean Wavpack_Tag_Write_File_Tag (ET_File *ETFile)
 {
-    if (!ETFile || !ETFile->FileTag)
-        return FALSE;
-
     WavpackContext *wpc;
 
     gchar    *filename = ((File_Name *)((GList *)ETFile->FileNameCur)->data)->value;
@@ -282,6 +279,9 @@ gboolean Wavpack_Tag_Write_File_Tag (ET_File *ETFile)
 
     int open_flags = OPEN_EDIT_TAGS;
 
+    if (!ETFile || !ETFile->FileTag)
+        return FALSE;
+
     wpc = WavpackOpenFileInput(filename, NULL, open_flags, 0);
 
     if ( wpc == NULL ) {
--
cgit v0.9.1
